src/flamegraph/flamegraph.js

```diff
@@ -158,29 +205,29 @@ function zoom_child(e, x, ratio) {
     if (e.attributes != undefined) {
         if (e.attributes.x != undefined) {
             orig_save(e, "x");
-            e.attributes.x.value = (parseFloat(e.attributes.x.value) - x - xpad) * ratio + xpad;
+            e.attributes.x.value = format_percent((parseFloat(e.attributes.x.value) - x) * ratio);
             if(e.tagName == "text")
-                e.attributes.x.value = find_child(e.parentNode, "rect[x]").attributes.x.value + 3;
+                e.attributes.x.value = format_percent(parseFloat(find_child(e.parentNode, "rect[x]").attributes.x.value) + 3);
```

Comment:
This should probably not be `+3`?

Reply:
Right, the 3 pixels should be converted to a percent here.